controlplane/kubeadm/internal/cluster.go

```diff
@@ -374,6 +418,15 @@ func newClientCert(caCert *x509.Certificate, key *rsa.PrivateKey, caKey *rsa.Pri
 	return c, errors.WithStack(err)
 }
 
-func etcdStaticPodName(nodeName string) string {
-	return "etcd-" + nodeName
+func staticPodName(component, nodeName string) string {
+	return fmt.Sprintf("%s-%s", component, nodeName)
+}
+
+func checkStaticPodReadyCondition(pod *corev1.Pod) error {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodReady && condition.Status != corev1.ConditionTrue {
```

Comment:
Is it guaranteed for a pod to always have the `PodReady` condition? If it's missing this returns nil

Reply:
ah, good catch. This should return err if it doesn't have a pod condition as that would indicate we don't know if it's ready or not and we want to err on the side of safe. All other decisions thus far have err'd on the side of safe with intentional improvements later on.